src/test/java/co/rsk/federate/FedNodeRunnerTest.java

```diff
@@ -0,0 +1,269 @@
+        SignerConfig rskSignerConfig = getRSKSignerConfig();
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.BTC_KEY_ID.getId())).thenReturn(btcSignerConfig);
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.RSK_KEY_ID.getId())).thenReturn(rskSignerConfig);
+
+        fedNodeRunner.run();
+
+        ECDSASigner signer = TestUtils.getInternalState(fedNodeRunner, "signer");
+        assertNotNull(signer);
+        assertTrue(signer.canSignWith(FedNodeRunner.BTC_KEY_ID));
+        assertTrue(signer.canSignWith(FedNodeRunner.RSK_KEY_ID));
+        assertFalse(signer.canSignWith(FedNodeRunner.MST_KEY_ID));
+
+        assertTrue(signer instanceof ECDSACompositeSigner);
+        ECDSACompositeSigner compositeSigner = (ECDSACompositeSigner) signer;
+        List<ECDSASigner> signers = TestUtils.getInternalState(compositeSigner, "signers");
+        assertEquals(2, signers.size());
+
+        HSMBookkeepingClient hsmBookkeepingClient = TestUtils.getInternalState(fedNodeRunner, "hsmBookkeepingClient");
+        assertNull(hsmBookkeepingClient);
+
+        HSMBookkeepingService hsmBookkeepingService = TestUtils.getInternalState(fedNodeRunner, "hsmBookkeepingService");
+        assertNull(hsmBookkeepingService);
+    }
+
+    @Test
+    public void test_KeyFile_config_with_no_path() throws Exception {
+        Config mockConfig = mock(Config.class);
+        when(mockConfig.getString("path")).thenReturn("");
+        SignerConfig btcSignerConfig = getBTCSignerConfig();
+        when(btcSignerConfig.getConfig()).thenReturn(mockConfig);
+        SignerConfig rskSignerConfig = getRSKSignerConfig();
+        when(rskSignerConfig.getConfig()).thenReturn(mockConfig);
+        SignerConfig mstSignerConfig = getMSTSignerConfig();
+        when(mstSignerConfig.getConfig()).thenReturn(mockConfig);
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.BTC_KEY_ID.getId())).thenReturn(btcSignerConfig);
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.RSK_KEY_ID.getId())).thenReturn(rskSignerConfig);
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.MST_KEY_ID.getId())).thenReturn(mstSignerConfig);
+
+        fedNodeRunner.run();
+
+        ECDSASigner signer = TestUtils.getInternalState(fedNodeRunner, "signer");
+        ECDSACompositeSigner compositeSigner = (ECDSACompositeSigner) signer;
+        List<ECDSASigner> signers = TestUtils.getInternalState(compositeSigner, "signers");
+        signers.forEach(keyFileSigner -> assertTrue(keyFileSigner instanceof ECDSASignerFromFileKey));
+        signers.forEach(keyFileSigner -> {
+            assertFalse(keyFileSigner.check().wasSuccessful());
+            assertTrue(keyFileSigner.check().getMessages().contains("Invalid Key File Name"));
+            assertTrue(keyFileSigner.check().getMessages().contains("Invalid key file permissions"));
+        });
+    }
+
+    @Test
+    public void test_with_no_config() throws Exception {
```

Comment:
A test where one of the signers config is duplicated? Which one will it consider?

Reply:
It Will Consider The First Config